pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -901,32 +1072,32 @@ func shouldCompressResponse(request, response *dns.Msg) bool {
 	return false
 }
 
-func GetSelectorRegexMap(l7 policy.L7DataMap) (CachedSelectorREEntry, error) {
-	newRE := make(CachedSelectorREEntry)
-	for selector, l7Rules := range l7 {
-		if l7Rules == nil {
-			l7Rules = &policy.PerSelectorPolicy{L7Rules: api.L7Rules{DNS: []api.PortRuleDNS{{MatchPattern: "*"}}}}
+func GenerateRegexpAndFqdns(l7Rules *policy.PerSelectorPolicy) (string, []string) {
```

Comment:
I know there was no godoc on this before, but could we add one? I think you know this code better than most now :)

Reply:
Yee, fair point! Added it now 👌